### PR TITLE
Add `RHASH_SIZE`

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: src/ruby.rs [hunk 1]
```diff
@@ -608,6 +608,22 @@ extern {
...
+    ///
+    /// * `hash` - an instance of [`Hash`](rb_cHash)
+    ///
+    /// # Safety
+    ///
+    /// * Undefined behavior if `hash` is not a `Hash`
+    ///
+    /// # Miscellaneous
+    ///
+    /// The macro is currently redefined in `internal.h`.
```
[the remark this answers is not in the document]
Comment (chancancode) [reply src/ruby.rs]: Ah. I just misunderstood your comment. Looking at the [commit](https://github.com/ruby/ruby/commit/8250aa2df0d6b4dcfa11dbad5307d28c2d5dd85f), I believe it is just an implementation detail of the hash class. Basically, the public API version of `RHASH_SIZE` you will be using will be the one defined in ruby.h, so I think we can safely drop this.

[the remark this answers is not in the document]
Comment (chancancode) [reply src/ruby.rs]: (Specifically, it was moving things out of `includes` – which hoses the public-ish APIs into the private header files that you can't link to)

File: ext/ruby.c [hunk 2]
```diff
@@ -12,3 +12,7 @@ long RS_RSTRING_LEN(VALUE str) { return RSTRING_LEN(str); }
 const char* RS_RSTRING_PTR(VALUE str) { return RSTRING_PTR(str); }
 
 long RS_RARRAY_LEN(VALUE a) { return RARRAY_LEN(a); }
+
+// If the arch supports it, it's a ull, otherwise just a ulong.
+// Should be ok to do ull here.
```
[the remark this answers is not in the document]
Comment (chancancode) [reply ext/ruby.c]: There are apparently such platforms that _Ruby_ still supports (at least the code still support it). This is something we need a global policy/answer for and blocks #51 as well.

There are a couple of scenarios:

1. Ruby no longer support those platforms/compilers and the code is just bit-rotting. In which case we can safely ignore this every time we see it in the codebase, and should probably PR to ruby/ruby to clean things up.

2. Ruby still support them but we don't care to. In which case we should document that.

3. We both support them. In which case we need to find a better way to do this.

[the remark this answers is not in the document]
Comment (wagenet) [reply ext/ruby.c]: AFAICT, `long long` does exist on 32-bit C, so this should be safe.

**Comment by wagenet** [comment]
@chancancode I've addressed both of your concerns.

**Review by chancancode — CHANGES_REQUESTED** [review]
I think 
https://github.com/tildeio/libcruby-sys/pull/50#discussion_r217238422 is still an outstanding question. Ruby seems to think that there are supported C compilers in-use that would barf on `unsigned long long`, or otherwise have the "wrong" size that it needs to. (Also this may actually want to be a `libc::size_t`.) I self-assigned for the investigation.

**Review by chancancode — APPROVED** [review]
we confirmed this is supposed to return a `size_t`

**Pull request merged**